app/assets/src/components/views/amr_heatmap/AMRHeatmapView.jsx

```diff
@@ -185,10 +206,38 @@ export default class AMRHeatmapView extends React.Component {
     }
   }
 
+  getDownloadCSVLink(csvOption) {
+    const dataString = this.computeHeatmapValuesForCSV();
+    const dataBlob = new Blob([dataString], { type: "text/csv" });
```

Comment:
Would it be possible to factor some of this into a separate method `createCSVObjectURL` in `utils/csv.js`?

Maybe `createCSVObjectURL`  can take the headers and rows that you compute in `computeHeatmapValuesForCSV` and handle the logic of creating the data string, turning it into a blob, and turning it into an object URL. This feels like a method that could be useful in the future.